sprites/tagalong.asm

```diff
@@ -0,0 +1,1614 @@
+      LDA.w Tagalong_TriggerData_dungeon_coordinates+2, X : STA $02
+      LDA.w Tagalong_TriggerData_dungeon_coordinates+4, X : STA $06
+      LDA.w Tagalong_TriggerData_dungeon_coordinates+8, X : STA $04
+      
+      SEP #$30
+      
+      JSR Tagalong_CheckTextTriggerProximity : BCS .check_flags_and_proximity
+      
+      REP #$30
+    
+    .not_room_data_match
+    
+    LDA $0C : CLC : ADC.w #$000A : TAX
+    
+    CPX $08 : BNE .next_room_data_block
+    
+    BRL .return
+    
+    .area_match
+    
+    LDA.w Tagalong_TriggerData_area_data_boundaries_1+2, X : STA $08
+    LDA.w Tagalong_TriggerData_area_data_boundaries_1, X : TAX
+    
+    .next_area_data_block
+    
+    STX $0C
+    STZ $0A
+    
+    LDA $7EF3CC : AND.w #$00FF : CMP $A55C, X : BNE .not_area_data_match
+    
+    LDA.w Tagalong_TriggerData_area_data_1, X : STA $00
+    LDA.w Tagalong_TriggerData_area_data_1+2, X : STA $02
+    LDA.w Tagalong_TriggerData_area_data_1+6, X : STA $06
+    LDA.w Tagalong_TriggerData_area_data_1+8, X : STA $04
+    
+    SEP #$30
+    
+    JSR Tagalong_CheckTextTriggerProximity : BCS .check_flags_and_proximity
+    
+    REP #$30
+    
+    .not_area_data_match
+    
+    LDA $0C : CLC : ADC.w #$000A : TAX
+    
+    CPX $08 : BNE .next_area_data_block
+    
+    BRA .return
+    
+    .check_flags_and_proximity
+    
+    SEP #$10
+    REP #$20
+    
+    ; Message has already triggered once during this instance of being
+    ; in this room or area.
+    LDA $02F2 : AND $06 : BNE .return
+    
+    LDA $06 : TSB $02F2
+    
+    ; Configure the text message index.
+    LDA $04 : STA $1CF0
+    
+    CMP.w #$FFFF : BEQ .no_text_message
+    CMP.w #$009D : BEQ .certain_kiki_message
+    CMP.w #$0028 : BNE .show_text_message
+    
+    SEP #$20
+    
+    LDA #$00 : STA $7EF3CC
+    
+    BRA .show_text_message
+    
+    .certain_kiki_message
+    
+    SEP #$20
+    
+    LDA $02CF : INC A : CMP.b #$14 : BNE .tagalong_state_index_not_maxed
+    
+    LDA.b #$00
+    
+    .tagalong_state_index_not_maxed
+    
+    JSL OldMountainMan_TransitionFromTagalong
+    
+    .show_text_message
+    
+    SEP #$20
+    
+    JSL Main_ShowTextMessage
+    
+    BRA .return
+    
+    .no_text_message
+    
+    SEP #$30
+    
+    LDA $02CF : INC A : CMP.b #$14 : BNE .tagalong_state_index_not_maxed_2
+    
+    LDA.b #$00
+    
+    .tagalong_state_index_not_maxed_2
+    
+    PHA
+    
+    LDA $06 : AND.b #$03 : BNE .kiki_first_begging_sequence
+    
+    PLA
+    
+    JSL Kiki_InitiatePalaceOpeningProposal
+    
+    BRA .return
+    
+    .kiki_first_begging_sequence
+    
+    PLA : STA $00
+    
+    LDX $8A
+    
+    LDA $7EF280, X : AND.b #$01 : BNE .return
+    
+    LDA $00
+    
+    JSL Kiki_InitiateFirstBeggingSequence
+    
+    .return
+    
+    SEP #$30
+    
+    RTS
+}
+
+; ==============================================================================
+
+; $04A6CD-$04A906 DATA
+TagalongDraw_Drawing:
+{    
+  .props
```

Comment:
labels for data should be on the same indentation as the data as well

Reply:
```
    .props
    db $20, $C0, $00 ; up
    db $00, $A0, $00 ; down
    db $40, $60, $00 ; left
    db $40, $60, $44 ; right
```

Reply:
Done